tests/unit/ceed/test_ceed.cpp

```diff
@@ -0,0 +1,199 @@
+
+   H1_FECollection fec(order, dim);
+   bool vecOp = pb == Problem::VectorMass || pb == Problem::VectorDiffusion;
+   const int vdim = vecOp ? dim : 1;
+   FiniteElementSpace fes(&mesh, &fec, vdim);
+
+   BilinearForm k_test(&fes);
+   BilinearForm k_ref(&fes);
+
+   FiniteElementSpace coeff_fes(&mesh, &fec);
+   GridFunction gf(&coeff_fes);
+   FunctionCoefficient f_coeff(coeff_function);
+
+   Coefficient *coeff = nullptr;
+   switch (coeff_type)
+   {
+      case CeedCoeff::Const:
+         coeff = new ConstantCoefficient(1.0);
+         break;
+      case CeedCoeff::Grid:
+         gf.ProjectCoefficient(f_coeff);
+         coeff = new GridFunctionCoefficient(&gf);
+         break;
+      case CeedCoeff::Quad:
+         coeff = &f_coeff;
+         break;
+   }
+
+   switch (pb)
+   {
+   case Problem::Mass:
+      k_ref.AddDomainIntegrator(new MassIntegrator(*coeff));
+      k_test.AddDomainIntegrator(new MassIntegrator(*coeff));
+      break;
+   case Problem::Diffusion:
+      k_ref.AddDomainIntegrator(new DiffusionIntegrator(*coeff));
+      k_test.AddDomainIntegrator(new DiffusionIntegrator(*coeff));
+      break;
+   case Problem::VectorMass:
+      k_ref.AddDomainIntegrator(new VectorMassIntegrator(*coeff));
+      k_test.AddDomainIntegrator(new VectorMassIntegrator(*coeff));
+      break;
+   case Problem::VectorDiffusion:
+      k_ref.AddDomainIntegrator(new VectorDiffusionIntegrator(*coeff));
+      k_test.AddDomainIntegrator(new VectorDiffusionIntegrator(*coeff));
+      break;
+   }
+
+   k_ref.Assemble();
+   k_ref.Finalize();
+
+   k_test.SetAssemblyLevel(assembly);
+   k_test.Assemble();
+
+   GridFunction x(&fes), y_ref(&fes), y_test(&fes);
+
+   x.Randomize(1);
+
+   k_ref.Mult(x,y_ref);
+   k_test.Mult(x,y_test);
+
+   y_test -= y_ref;
+
+   REQUIRE(y_test.Norml2() < 1.e-12);
+}
+
+TEST_CASE("CEED", "[CEED]")
+{
+   auto assembly = GENERATE(AssemblyLevel::PARTIAL,AssemblyLevel::NONE);
+   auto coeff_type = GENERATE(CeedCoeff::Const,CeedCoeff::Grid,CeedCoeff::Quad);
+   auto pb = GENERATE(Problem::Mass,Problem::Diffusion,
+                      Problem::VectorMass,Problem::VectorDiffusion);
+   auto order = GENERATE(1,2,4);
+   auto mesh = GENERATE("../../data/inline-quad.mesh","../../data/inline-hex.mesh",
+                        "../../data/star-q3.mesh","../../data/fichera-q3.mesh",
+                        "../../data/amr-quad.mesh","../../data/fichera-amr.mesh");
+   test_ceed_operator(mesh, order, coeff_type, pb, assembly);
+} // test case
+
+} // namespace ceed_test
+
+int main(int argc, char *argv[])
+{
+   // There must be exactly one instance.
+   Catch::Session session;
+
+   const char *device_str = (argc == 1) ? "ceed-cpu" : argv[argc-1];
+
+   // Apply provided command line arguments.
+   int r = session.applyCommandLine((argc == 1) ? argc : argc - 1, argv);
+   if (r != 0)
+   {
+      return r;
+   }
+
+   Device device(device_str);
+
+   int result = session.run();
+
+   return result;
+}
```

Comment:
```suggestion
int main(int argc, char *argv[])
{
   // There must be exactly one instance.
   Catch::Session session;
   std::string device_str("ceed-cpu");
   using namespace Catch::clara;
   auto cli = session.cli()
      | Opt(device_str, "device_string")
        ["--device"]
        ("CEED device string (default: ceed-cpu)");
   session.cli(cli);
   int result = session.applyCommandLine( argc, argv );
   if (result != 0)
   {
      return result;
   }
   Device device(device_str.c_str());
   result = session.run();
   return result;
}
```

Now that we are using the new version of Catch, we can do something like this to customize the command line arguments.

Reply:
Thank you, it's very nice indeed!